BASE/src/Microsoft.ApplicationInsights/Extensibility/Implementation/InstrumentationKeyValidation.cs

```diff
@@ -0,0 +1,40 @@
+﻿namespace Microsoft.ApplicationInsights.Extensibility.Implementation
+{
+    using System;
+    using System.Text.RegularExpressions;
+
+    internal static class InstrumentationKeyValidation
+    {
+        /// <summary>
+        /// Checks if an instrumentation key contains invalid characters.
+        /// </summary>
+        /// <param name="instrumentationKey">Candidate instrumentation key.</param>
+        /// <exception cref="ArgumentNullException">Throws if instrumentation key is null.</exception>
+        /// <exception cref="ArgumentException">Throws if instrumentation key is invalid with description.</exception>
+        public static void Validate(string instrumentationKey)
+        {
+            if (instrumentationKey == null)
+            {
+                throw new ArgumentNullException(nameof(instrumentationKey));
+            }
+            else if (string.IsNullOrWhiteSpace(instrumentationKey))
+            {
+                throw new ArgumentException("Instrumentation Key can not be empty.");
+            }
+
+            Regex allowedCharacters = new Regex("^[a-zA-Z0-9-:]*$");
```

Comment:
This excludes Unicode alphanumeric character and only includes English alphanumeric - do we expect IKey string to be non-English in some cases? If yes, we can change the Regex to account for that.